sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/HadoopInputFormatIOTest.java

```diff
@@ -0,0 +1,776 @@
+    SerializableConfiguration serConf =
+      loadTestConfiguration(
+          BadNullCreateReaderInputFormat.class, 
+          Text.class, 
+          Employee.class);
+    HadoopInputFormatBoundedSource<Text, Employee> parentHIFSource =
+        new HadoopInputFormatBoundedSource<Text, Employee>(
+            serConf, 
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class));
+    List<BoundedSource<KV<Text, Employee>>> boundedSourceList = 
+        (List<BoundedSource<KV<Text, Employee>>>) parentHIFSource
+              .splitIntoBundles(0, p.getOptions());
+    for (BoundedSource<KV<Text, Employee>> source : boundedSourceList) {
+      BoundedReader<KV<Text, Employee>> reader = source.createReader(p.getOptions());
+      thrown.expect(IOException.class);
+      thrown.expectMessage(
+          String.format(HadoopInputFormatIOContants.NULL_CREATE_RECORDREADER_ERROR_MSG,
+              new BadNullCreateReaderInputFormat().getClass()));
+      reader.start();
+    }
+  }
+
+  /**
+   * This test validates behavior of createReader() and start() methods if InputFormat's getSplits()
+   * returns InputSplitList having having no records.
+   */
+  @Test
+  public void testReadersCreateReaderAndStartWithZeroRecords() throws Exception {
+    SerializableConfiguration serConf = 
+     loadTestConfiguration(
+        BadNoRecordsInputFormat.class,
+        java.lang.String.class, 
+        java.lang.String.class);
+    HadoopInputFormatBoundedSource<Text, Employee> parentHIFSource =
+        new HadoopInputFormatBoundedSource<Text, Employee>(
+            serConf, 
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class));
+    List<BoundedSource<KV<Text, Employee>>> boundedSourceList =
+        (List<BoundedSource<KV<Text, Employee>>>) parentHIFSource
+              .splitIntoBundles(0, p.getOptions());
+    for (BoundedSource<KV<Text, Employee>> source : boundedSourceList) {
+      BoundedReader<KV<Text, Employee>> reader = source.createReader(p.getOptions());
+      assertEquals(false, reader.start());
+    }
+  }
+
+  /**
+   * This test validates fractions consumed while reading data.
+   */
+  @Test
+  public void testReadersGetFractionConsumed() throws Exception {
+    List<KV<Text, Employee>> referenceRecords = UnitTestUtils.getEmployeeData();
+    HadoopInputFormatBoundedSource<Text, Employee> parentHIFSource =
+        new HadoopInputFormatBoundedSource<Text, Employee>(
+            serConf, 
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class));
+    long estimatedSize = parentHIFSource.getEstimatedSizeBytes(p.getOptions());
+    assertEquals(referenceRecords.size(), estimatedSize);
+    List<BoundedSource<KV<Text, Employee>>> boundedSourceList =
+        (List<BoundedSource<KV<Text, Employee>>>) parentHIFSource
+            .splitIntoBundles(0, p.getOptions());
+    assertEquals(UnitTestUtils.NUMBER_OF_SPLITS, boundedSourceList.size());
```

Comment:
"UnitTestUtils" can be a catch-all bucket for anything related to this unit test (which is a an anti-pattern - http://stackoverflow.com/questions/3340032/utility-classes-are-evil)

The Utils class you have is actually just fine since it doesn't have a bucket of random unrelated code, I just think the UnitTestUtils class could be renamed TestEmployeeDataSet or something like that to reflect what it is actually doing.